ivy_tests/test_ivy/test_frontends/test_tensorflow/test_nn.py

```diff
@@ -1451,3 +1451,52 @@ def test_tensorflow_avg_pool3d(
         strides=strides,
         padding=padding,
     )
+
+
+@st.composite
+def _pool_args(draw):
+    dims = draw(st.integers(min_value=1, max_value=3))
+    data_formats = ["NWC", "NHWC", "NDHWC"]
+    data_format = data_formats[dims - 1]
+    x = draw(
+        helpers.arrays_for_pooling(
+            min_dims=dims + 2, max_dims=dims + 2, min_side=1, max_side=4
+        )
+    )
+
+    dilation = draw(st.sampled_from([None, [1] * dims, [2] * dims]))
+    window_size = [
+        draw(st.integers(min_value=1, max_value=x[0].shape[i + 1]) | st.just(1)) for _ in range(dims)
```

Comment:
```
E   AttributeError: 'list' object has no attribute 'shape'
```

Reply:
Hi, I'm facing some errors while writing the pool function. I've given the full description of my problem in the commit summary, if you require anymore information please contact me.

Thank you for helping me